### PR TITLE
Object based image analysis using matlab

### LINKED ISSUE #4204: Object-based image analysis using Matlab
#### Introduction
Object-based image analysis is the processing of an image based on the classification of its pixels to get more useful information on the image, such information can be based on height, edges, and boundaries. Matlab provides an interactive environment for object-based image analysis through the execution of functions used in analysis or inbuilt apps for image processing.

Images contain objects with distinct regions, these regions have features such as area, perimeter, shapes, and height. Matlab allows for analysis of these properties using image analyzer functions or region props to obtain data from these images.

Object-based image analysis is useful especially in analyzing satellite maps, machine vision, and obtaining information based on object characteristics in an image.

#### Key takeaways 
- Distinguishing objects in an image using Matlab.
- Obtaining image object properties in Matlab.
- Classification of images based on image objects.
- Using region props in image analysis.

### PR DESCRIPTION
**Pull request by alpho361**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #4204

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by louisefindlay23** [comment]
Hi, @alpho361. Welcome to the EngEd program. 

To get your article in a ready to review state, please rename your article to index.md not index,md and your image file extensions to lowercase

Thanks,
Louise - Content Moderator

**Comment by alpho361** [comment]
@louisefindlay23 Changes made.

**Comment by louisefindlay23** [comment]
Thanks, @alpho361 but you also need to rename the image extensions themselves (in the image filenames) as well as the index.md file.

**Comment by alpho361** [comment]
@louisefindlay23 The extension (PNG) is by default, i have tried changing it in the image filename and it seems unchangeable.

**Comment by louisefindlay23** [comment]
Images can only be renamed using the command line. See this [tutorial](https://docs.github.com/en/repositories/working-with-files/managing-files/renaming-a-file#renaming-a-file-using-the-command-line) for instructions.

**Comment by alpho361** [comment]
@louisefindlay23  Changes made.

**Comment by louisefindlay23** [comment]
Thanks, @alpho361. Your article is now ready for review. One of our peer reviewers will review your article as soon as one becomes available.

**Comment by hectorkambow** [comment]
@section-engineering-education/peer-reviewers 

If anyone is available to review this one after they finish up any others in the queue. 👍

**Comment by jumamiller** [comment]
@alpho361, a nice article, but you need to add more value by explaining each code snippet to help readers follow along!

**Comment by ahmadmardeni1** [comment]
Plagiarism check done ✅

**Comment by alpho361** [comment]
@jumamiller Changes mede

File: content/articles/object-based-image-analysis-using-matlab/index.md [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+### Edge detection 
+Edge detection is the identification of points within an image. These points are where the image has excellent contrast, and they are the defining points of the image. Edge detection work by detecting changes in brightness of the image pixels.
+
+Edge detection is useful in image segmentation and data extraction for comparison, objects separation, computer vision, and machine learning. Matlab supports a variety of functions that aids in edge detection.
+
+### Methods of edge detection
+There are several methods used in edge detection in images. These methods are used with the primary function `edge`.  We will demonstrate most of these methods using a similar image and then compare the outcome to find a more suitable way to provide a more detailed output.
+
+All these methods only accept a grayscale image input; hence it is essential to fast convert the RGB image to a grayscale image, conversion of RGB to gray is done using the function `rgb2gray`. The initial stage of all image processes is importing the image to the Matlab workspace. Importation of image is done using the function `imread('imagefolderpath')`. 
+
```
Comment (jumamiller) [inline content/articles/object-based-image-analysis-using-matlab/index.md:36]: Ensure all functions have opening and closing brackets, e.g `rgb2gray()`. Edit where applicable throughout the article!

File: content/articles/object-based-image-analysis-using-matlab/index.md [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+![Grayscale version of the imported image](/engineering-education/object-based-image-analysis-using-matlab/objectanalysis_thirteen.jpg)
+
+```matlab
+bw1 = imbinarize(i2); % binarizing image
+figure,imshow(bw1)
+```
+![Binary image](/engineering-education/object-based-image-analysis-using-matlab/objectanalysis_fourteen.jpg)
+
+```matlab
+se = strel('disk',5); % specyfying strel lenght
```
Comment (jumamiller) [inline content/articles/object-based-image-analysis-using-matlab/index.md:137]: Have you explained all these snippets? Kindly explain all codes in this article!!

**Review by jumamiller — CHANGES_REQUESTED** [review]
Kindly explain all code snippets, it's important to let readers know what's going on.

**Comment by alpho361** [comment]
@jumamiller Changes made.

**Comment by jumamiller** [comment]
> @jumamiller Changes made.

I asked you to explain code snippets for the third time and you've not updated. I'll close this article for now. When you have finished explaining all the snippets, let me know to reopen the article!

**Comment by alpho361** [comment]
@jumamiller Changes made.

**Comment by jumamiller** [comment]
Opening this article to check for latest commits as pointed above!

File: content/articles/object-based-image-analysis-using-matlab/index.md [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+figure, imshow(l) %displaying edges detected using log method
+```
+![Edge detected using log method](/engineering-education/object-based-image-analysis-using-matlab/objectanalysis_five.jpg)
+
+- Canny: This method detects the edges by separating noise from the image. This is better because it does not disturb the features of the edges in the image. This method uses the function `canny()` and the syntax is `i_edges = edge(i_gray,'canny')`.
+ The below codes shows edge detection using the canny method.
+```matlab
+c = edge(i2,'canny'); %edge detection using canny method
+figure, imshow(c) %displaying edges detected using canny method
+```
```
Comment (jumamiller) [inline content/articles/object-based-image-analysis-using-matlab/index.md:84]: Hello, what does code do? why is the `edge(i2,'canny')`, described this way? I've requested this severally @alpho361!

**Review by jumamiller — CHANGES_REQUESTED** [review]
@alpho361, why don't you reach out for help? Explaining a code is simply walking the reader through the code to understand what you're doing!
For example, when I've got a method 'createUser(Request $request)`, I would say, for instance, "the above method createUser(...), takes in a single paramater request that......."

**Comment by jumamiller** [comment]
> @jumamiller Changes made.

Not even a single code explained? did you follow the requirements?

**Comment by alpho361** [comment]
@jumamiller  Changes made.

**Comment by jumamiller** [comment]
@alpho361, kindly work on the grammar. See the above commits to improve your articles in the future.

File: content/articles/object-based-image-analysis-using-matlab/index.md [hunk 4]
```diff
@@ -0,0 +1,241 @@
+
+### Introduction
+Object-based image analysis is the processing of an image based on the classification of its pixels to get more helpful information on the image. For example, such data can be based on height, edges, and boundaries. Matlab provides an interactive environment for object-based image analysis by executing functions used in analysis or inbuilt apps for image processing.
+
```
Comment (jumamiller) [inline content/articles/object-based-image-analysis-using-matlab/index.md:4]: what do you mean by 'For example, such data......". Give a precise description of what you're covering.

File: content/articles/object-based-image-analysis-using-matlab/index.md [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+Images contain objects with distinct regions. These regions have area, perimeter, shapes, and height features. Matlab allows for analysis of these properties using image analyzer functions or region props to obtain data from these images.
+
+Object-based image analysis is useful, especially in analyzing satellite maps, machine vision, and obtaining information based on object characteristics in an image.
```
Comment (jumamiller) [inline content/articles/object-based-image-analysis-using-matlab/index.md:7]: Generall, tell us what this article will cover. Kindly go thro' past articles to get more info about introductions!

File: content/articles/object-based-image-analysis-using-matlab/index.md [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+
+
+### Edge detection 
+Edge detection is the identification of points within an image. These points are where the image has excellent contrast, and they are the defining points of the image. Edge detection work by detecting changes in brightness of the image pixels.
+
+Edge detection is useful in image segmentation and data extraction for comparison, objects separation, computer vision, and machine learning. Matlab supports a variety of functions that aids in edge detection.
+
+### Methods of edge detection
+There are several methods used in edge detection in images. These methods are used with the primary function `edge()`.  We will demonstrate most of these methods using a similar image and then compare the outcome to find a more suitable way to provide a more detailed output.
+
```
Comment (jumamiller) [inline content/articles/object-based-image-analysis-using-matlab/index.md:34]: Could you refrain from "we will". It's almost everywhere in the article!

File: content/articles/object-based-image-analysis-using-matlab/index.md [hunk 7]
```diff
@@ -0,0 +1,241 @@
...
+All these methods only accept a grayscale image input; hence, converting the RGB image to a grayscale image is essential. RGB to gray conversion is converted using the function `rgb2gray()`. The initial stage of all image processes is importing the image to the Matlab workspace. Importation of image is done using the function `imread('imagefolderpath')`. 
+
+The following codes are used in importation and image conversion to grayscale. The outcome will be used to demonstrate various edge detection methods.
+
+First, we import the image using the function `imread()` as demonstrated with the below codes.
```
Comment (jumamiller) [inline content/articles/object-based-image-analysis-using-matlab/index.md:39]: Have these explanations after the code. The current way is affecting the general article flow!

**Review by jumamiller — CHANGES_REQUESTED** [review]
@alpho361, I think this article is still far behind to be published. Please take your time, and work on a ready to publish article.

- I've fixed multiple grammar mistakes, but it seems the general flow ain't there. Read previous articles to understand how you can keep your readers glued on the article.

**Comment by jumamiller** [comment]
I'll close for now untill all changes are done!

**Comment by alpho361** [comment]
@jumamiller Can you please reopen the PR, have made changes.

**Comment by lalith1403** [comment]
Can you create a new PR and ping the peer reviewer. Thanks.

**Comment by jumamiller** [comment]
> @jumamiller Can you please reopen the PR, have made changes.

reopening to reflect commits

**Review by jumamiller — APPROVED** [review]
@alpho, you have done great work reworking on this article.

**Comment by hectorkambow** [comment]
jumping on this one

**Comment by hectorkambow** [comment]
👍 overall nice work - very detailed and straight forward tutorial 
 
Please review and suggestions/changes made by your peer reviewer and/or myself during a final review.

**Pull request merged**